docstore/docstore.go

```diff
@@ -338,15 +370,42 @@ func (l *ActionList) Do(ctx context.Context) error {
 	return l.do(ctx, true)
 }
 
-// do implements Do with optional OpenCensus tracing, so it can be used internally.
-func (l *ActionList) do(ctx context.Context, oc bool) (err error) {
+// do implements Do with optional OpenTelemetry tracing, so it can be used internally.
+func (l *ActionList) do(ctx context.Context, withTracing bool) (err error) {
 	if err := l.coll.checkClosed(); err != nil {
 		return ActionListError{{-1, errClosed}}
 	}
 
-	if oc {
-		ctx = l.coll.tracer.Start(ctx, "ActionList.Do")
-		defer func() { l.coll.tracer.End(ctx, err) }()
+	var startTime time.Time
+	var span trace.Span
+
+	if withTracing {
+		startTime = time.Now()
+		ctx, span = l.coll.tracer.Start(ctx, "ActionList.Do")
+		defer func() {
+			l.coll.tracer.End(span, err)
+
+			// Record metrics
+			if completedCallsCounter != nil {
+				attr := []attribute.KeyValue{
+					gcdkotel.MethodKey.String("ActionList.Do"),
+					gcdkotel.ProviderKey.String(l.coll.tracer.Provider),
+				}
+				completedCallsCounter.Add(context.Background(), 1, metric.WithAttributes(attr...))
```

Comment:
Why not `ctx`?

Reply:
@jba This has been resolved appropriately